components/micro_wake_word.rst

```diff
@@ -0,0 +1,106 @@
+Micro Wake Word
+================
+
+.. seo::
+    :description: Instructions for creating a custom wake word using microWakeWord.
+    :image: voice-assistant.svg
+
+ESPHome implements an on-device wake word detection framework from `microWakeWord <https://github.com/kahrendt/microWakeWord>`__.
+This repository/library allows you to create a custom wake word for your ESPHome device.
+
+The training process is described on the `microWakeWord GitHub repository <https://github.com/kahrendt/microWakeWord>`__.
+
+The ``micro_wake_word`` component requires an **ESP32-S3** to function.
+
+.. code-block:: yaml
+
+    micro_wake_word:
+      model: hey_jarvis
+
+    micro_wake_word:
+      model: github://esphome/micro-wake-word-models/models/hey_jarvis.json
+
+Configuration variables:
+------------------------
+
+- **model** (**Required**, string): The model to use. This can be one of:
+
+  - A simple name of a model that exists in the official `ESPHome Models repository <https://github.com/esphome/micro-wake-word-models>`__.
+    e.g. ``hey_jarvis``.
+  - A github shorthand URL to a model JSON file.
+    e.g. ``github://esphome/micro-wake-word-models/models/hey_jarvis.json@main``.
+  - A full URL to a model JSON file.
+    e.g. ``https://github.com/esphome/micro-wake-word-models/raw/main/models/hey_jarvis.json``.
+
+- **on_wake_word_detected** (Optional, Automation): An automation to perform when the wake word is detected.
```

Comment:
Why is this optional? What use is micro wake word when it doesn't act on the wake word ? 

We should also add this to the example

Reply:
Triggers are always optional in ESPHome. Maybe you just wanna test it, in which case, it will log when detected.